hearts/game/manager.py

```diff
@@ -25,10 +26,12 @@ class GameManager:
     # The current trick in the deal.
     _current_trick: Optional[Trick] = None
 
+    @atomic
     def __init__(self, game: Game):
         """Initialize the game manager with a Game object."""
-        # Make the Game object accessible to instances of this class.
-        self.game = game
+        # Make the Game object accessible to instances of this class. Acquire
+        # lock on game object to prevent concurrent game updates.
+        self.game = Game.objects.select_for_update(nowait=True).get(id=game.id)
```

Comment:
This change is unrelated to notifications but it's used to help prevent concurrent calls. By using `select_for_update` on the game object, we acquire a lock that we hold onto until this class is exited. If another request comes in while the lock is acquired, an exception will be thrown because the lock is already held by the first request.

The `atomic` decorator is a requirement of `select_for_update`. Practically it means that any database changes in this method will get rolled back if an exception is raised